src/frontend/components/message.js

```diff
@@ -7,352 +7,287 @@ import { CookieService } from "../services/cookie-service";
+		}
+		.receiver::after {
+			content: "";
+			position: absolute;
+			top: 0px;
+			left: -9px;
+			border-top: 10px solid white;
+			border-left: 10px solid transparent;
+			border-right: 0px solid transparent;
+			z-index: 3;
+		}
+		.receiver::before {
+			content: "";
+			position: absolute;
+			top: -1px;
+			left: -13px;
+			border-top: 11px solid rgb(209 209 209 / 34%);
+			border-right: 0px solid transparent;
+			border-left: 12px solid transparent;
+			filter: blur(0.8px);
+			z-index: 2;
+		}
+		.receiver-name {
+			font-size: 13px;
+			color: blue;
+		}
+
+		.message {
+			overflow-wrap: break-word;
+		}
+
+		.settings-container {
+			position: relative;
+			background: white;
+			border-radius: 6px;
+			box-shadow: rgba(0, 0, 0, 0.16) 0px 1px 4px;
+			opacity: 0;
+			transition: opacity 0.5s;
+		}
+
+		.message-body:hover .settings-container {
+			opacity: 1;
+		}
+
+		.sender .message-timestamp {
+			text-align: end;
+			font-size: 11px;
+			color: #e9e9e9;
+		}
+
+		.sender ~ .settings-container il-message-settings {
+			position: absolute;
+			top: 0px;
+			left: -89px;
+		}
+
+		.receiver ~ .settings-container il-message-settings {
+			position: absolute;
+			top: 0px;
+			right: -144px;
+		}
+
+		.receiver .message-timestamp {
+			text-align: end;
+			font-size: 11px;
+			color: #8c8d8d;
+		}
+
+		.message-date {
+			justify-self: center;
+			padding: 5px;
+			border-radius: 6px;
+			background-color: rgb(221, 221, 221);
+		}
+	`;
+
+	render() {
+		return html`
+			${this.compareMessageDate(
+				this.messages[this.index - 1]?.timestamp,
+				this.message.timestamp
+			)}
+
+			<!-- da refactorizzare in una storia apposita -->
+
+			<div class="message-body">
+				<!--  message content -->
+				<div
+					class=${this.message.sender == this.cookie.username
+						? "sender"
+						: "receiver"}
+				>
+					<p class="receiver-name">
+						${this.message.sender != this.cookie.username
+							? this.message.sender
+							: ""}
+					</p>
+					<p class="message">
+						${resolveMarkdown(
+							MarkdownService.parseMarkdown(this.message.content)
+						)}
+					</p>
+					<p class="message-timestamp">
+						${new Date(this.message.timestamp).toLocaleTimeString([], {
+							hour: "2-digit",
+							minute: "2-digit",
+						})}
+					</p>
+				</div>
+
+				<!-- end -->
+				<!-- menu icon -->
+
+				<div class="settings-container">
+					<il-button-icon
+						@click=${this.toggleSettings}
+						content="${IconNames.dotsHorizontal}"
+						color="black"
+					>
+					</il-button-icon>
+
+					<il-message-settings
+						.message=${this.message}
+						.cookie=${this.cookie}
+						.index=${this.index}
+						.type=${this.message.sender == this.cookie.username
+							? "sender"
+							: "receiver"}
+					>
+					</il-message-settings>
+				</div>
+
+				<!-- end -->
+			</div>
+		`;
+	}
+
+	toggleSettings() {
+		let settings = this.renderRoot.querySelector("il-message-settings");
+
+		settings.openDialog();
+	}
```

Comment:
Old:
Gestirei la cosa con javascript togliendo l'hover, in modo che cliccando sui tre punti si apra il menù.

New:
L'apertura del menu-settings è regolata attraverso questa funzione